smartmin/views.py

```diff
@@ -476,9 +476,10 @@ def get_cancel_url(self):
         return smart_url(self.cancel_url, self.object)
 
     def pre_delete(self, obj):
-        # auto populate modified_by if it is present
-        if hasattr(obj, 'modified_by_id') and self.request.user.id >= 0:
-            obj.modified_by = self.request.user
+        if self.request.user.id is not None:
+            # auto populate modified_by if it is present
+            if hasattr(obj, 'modified_by_id') and self.request.user.id >= 0:
+                obj.modified_by = self.request.user
```

Comment:
I think you can simplify as `if self.request.user.id and hasattr(obj, 'modified_by_id')`, no?

Reply:
was not sure if we request.user.id == 0 is a special use case ? so left that code unchanged

Reply:
Actually I just looked on prod and `AnonymousUser` is in there with id `-1` - so I think the intention here was to exclude them. So maybe...

```python
if self.request.user.id and self.request.user.id > 0 and hasattr(obj, 'modified_by_id')
```